src/test/java/uk/gov/companieshouse/api/accounts/service/impl/ApprovalServiceTest.java

```diff
@@ -0,0 +1,134 @@
+import uk.gov.companieshouse.api.accounts.model.entity.ApprovalEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.Approval;
+import uk.gov.companieshouse.api.accounts.repository.ApprovalRepository;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transformer.ApprovalTransformer;
+import uk.gov.companieshouse.api.accounts.utility.impl.KeyIdGenerator;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(Lifecycle.PER_CLASS)
+public class ApprovalServiceTest {
+
+
+    @Mock
+    private Approval approval;
+
+    @Mock
+    private Transaction transaction;
+
+    @Mock
+    private ApprovalRepository approvalRepository;
+
+    @Mock
+    private SmallFullService smallFullService;
+
+    @Mock
+    private MessageDigest messageDigest;
+
+    @Mock
+    private ApprovalEntity approvalEntity;
+
+    @Mock
+    private ApprovalTransformer approvalTransformer;
+
+    @Mock
+    private DuplicateKeyException duplicateKeyException;
+
+    @Mock
+    private MongoException mongoException;
+
+    @Mock
+    private KeyIdGenerator keyIdGenerator;
+
+    @InjectMocks
+    private ApprovalService approvalService;
+
+    public void setUpCreate() {
+    }
+
+    @Test
+    @DisplayName("Tests the successful creation of an Approval resource")
+    public void canCreateAnApproval() throws DataException {
+        setUpCreate();
+        when(approvalTransformer.transform(approval)).thenReturn(approvalEntity);
+        ResponseObject<Approval> result = approvalService
+            .create(approval, transaction, "", "");
+        assertNotNull(result);
+        assertEquals(approval, result.getData());
+    }
+
+    @Test
+    @DisplayName("Tests the duplicate key when creating an Approval resource")
+    public void createApprovalDuplicateKey() throws DataException {
+        setUpCreate();
+        doReturn(approvalEntity).when(approvalTransformer).transform(ArgumentMatchers
+            .any(Approval.class));
+        when(approvalRepository.insert(approvalEntity)).thenThrow(duplicateKeyException);
+        ResponseObject response = approvalService.create(approval, transaction, "", "");
+        assertNotNull(response);
+        assertEquals(response.getStatus(), ResponseStatus.DUPLICATE_KEY_ERROR);
+        assertNull(response.getData());
+    }
+
+    @Test
+    @DisplayName("Tests the mongo exception when creating an Approval")
+    void createApprovalMongoExceptionFailure() throws DataException {
+        setUpCreate();
+        doReturn(approvalEntity).when(approvalTransformer).transform(ArgumentMatchers
+            .any(Approval.class));
+        when(approvalRepository.insert(approvalEntity)).thenThrow(mongoException);
+        Executable executable = () -> {
+            approvalService.create(approval, transaction, "", "");
```

Comment:
Don't need to assign the executable here, can just do `assertThrows(DataException.class, approvalService.create(approval, transaction, "", ""));`

Reply:
`assertThrows` only takes an Executable for the second parameter `approvalService.create(approval, transaction, "", "")` does not satisfy this?